lychee-lib/src/utils/request.rs

```diff
@@ -126,16 +126,15 @@ fn truncate_source(source: &InputSource) -> InputSource {
 pub(crate) fn create(
     uris: Vec<RawUri>,
     input_content: &InputContent,
-    base: &Option<Base>,
-    extractor: &Option<BasicAuthExtractor>,
+    base: Option<&Base>,
+    extractor: Option<&BasicAuthExtractor>,
 ) -> HashSet<Request> {
-    let base = base
-        .clone()
-        .or_else(|| Base::from_source(&input_content.source));
+    let base_fallback = Base::from_source(&input_content.source);
```

Comment:
@mre This is the only change I'm unsure about. Not really happy with this. Creating the `base_fallback` even when we don't need it is quite ugly. But I couldn't get it to work with inlining it or moving it into the `or_else` closure, because of lifetime issues. Do you have another idea?

Reply:
I agree. We can always disable the lint and add a comment.